bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckWorkItem.java

```diff
@@ -223,50 +211,54 @@ private boolean currentCheckValid(CensusInstance censusInstance, List<Comment> c
                 String previousPRMetadata = substrings[0];
                 String previousIssueMetadata = (substrings.length > 1) ? substrings[1] : "";
 
-                // triggered by issue update
-                if (spawnedFromIssueBot) {
+                // triggered by issue update or initial run when bot restarts
+                if (initialRun || spawnedFromIssueBot) {
                     var currIssueMetadata = getIssueMetadata(pr.body());
                     if (expiresAt != null) {
                         if (previousIssueMetadata.equals(currIssueMetadata) && expiresAt.isAfter(Instant.now())) {
                             log.finer("[Issue]Metadata with expiration time is still valid, not checking again");
-                            return true;
```

Comment:
I think this method would be easier to read if we kept all return `true`/`false` inside the conditional blocks. If we know we have a terminal state, lets be clear about it so the reader doesn't need to check the rest of the method.

Reply:
I removed `return true` here because we couldn't return early in this block. For the initial run case, `initialRun` would be true and `spawnedFromIssueBot` would be false. We need to check both issueMetaData and PRMetaData. 

But we could add some `return true` in `if (!spawnedFromIssueBot)` block.

Reply:
Or I could switch the order of `if (initialRun || spawnedFromIssueBot)` and `if (!spawnedFromIssueBot)`.